app/tiebout.hoon

```diff
@@ -0,0 +1,253 @@
+    [ost.bol %hiss /request [~ ~] %httr %hiss (create-apns-request not)]~
+  [~ this]
+::
+::  +diff-hall-prize: receive new circle data
+::
+++  diff-hall-prize
+  |=  [wir=wire piz=prize:hall]
+  ^-  (quip move _this)
+  ~&  prize+[wir piz]
+  ?~  wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+  ?+  wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+  ::
+  ::  %our: set config of circle and iterate through messages, sending
+  ::  notifications for all messages where number is higher than our last-read
+  ::
+      {%our @ @}
+    ?>  ?=(%circle -.piz)
+    =/  nom/name:hall  i.t.wir
+    =/  red/@ud  red.loc.cos.piz
+    [~ this(circles.tie.sta (~(put by circles.tie.sta) nom red))]
+  ==
+::
+::  +reap: recieve acknowledgement for peer
+::
+++  reap
+  |=  [wir=wire err=(unit tang)]
+  ^-  (quip move _this)
+  ::~&  reap+[wir =(~ err)]
+  ?~  err
+    ::  XX send message to users inbox
+    [~ this]
+  ?~  wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+  ?+  i.wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+      %our
+    ?<  ?=(~ t.wir)
+    :_  this
+    ~
+  ==
+::
+::  +quit: receive subscription failed, resubscribe
+::
+++  quit
+  |=  wir=wire
+  ^-  (quip move _this)
+  ?~  wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+  ?+  i.wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+      %our
+    ?<  ?=(~ t.wir)
+    :_  this
+    =/  doc/dock  [our.bol %hall]
+    [ost.bol %peer /our/[i.t.wir] doc /circle/[i.t.wir]/config/grams]~
+  ==
+
+::
+::  +diff-hall-rumor: receive message or a read event from a hall circle
+::
+++  diff-hall-rumor
+  |=  [wir=wire rum=rumor:hall]
+  ^-  (quip move _this)
+  ::~&  rumor+[wir rum]
+  ?~  wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+  ?+  wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+  ::
+  ::  %our
+  ::
+      {%our @ @}
+    ?>  ?=(%circle -.rum)
+    =/  nom/name:hall  i.t.wir
+    ?+  -.rum.rum
+      [~ this]
+    ::
+    ::  %gram: send notification if envelope is lower than read number
+    ::
+      %gram
+    =/  red  (~(get by circles.tie.sta) nom)
+    ?~  red
+      (mean [leaf+"invalid circle for diff: {(spud wir)}"]~)
+    ?:  (gth num.nev.rum.rum u.red)
+      :_  this(circles.tie.sta (~(put by circles.tie.sta) nom u.red))
+      (message-to-notification u.red nev.rum.rum)
+    :_  this
+    (message-to-notification u.red nev.rum.rum)
+    ::
+    ::   %config: set our read number
+    ::
+      %config
+    ?+  -.dif.rum.rum
+      [~ this]
+          %read
+        =/  red/@ud  red.dif.rum.rum
+        [~ this(circles.tie.sta (~(put by circles.tie.sta) nom red))]
+      ==
+    ==
+  ==
+::
+::  generate notification move from hall message
+::
+++  message-to-notification
+  |=  [red=@ud env=envelope:hall]
+  ^-  (list move)
+  ?:  =(aut.gam.env our.bol)
+    ~
+  =/  pay  %-  my  :~
+    alert+s+'New message from {(cite:title aut.gam.env)}'
+  ==
```

Comment:
This could do with a type for `pay`. `my` isn't commonly used, `~(gas in *(map @t json))` is more verbose/clear on what it does (and type-checks early!). For this case, you could also `~(put by *(map @t json))` since you only have one element.

Or just do the dumbest thing and say `[alert+s+'etc' ~ ~]` and cast that to a `map`. (^:

(The most aggressive thing to do here, of course, is to tell @pilfer-pandex to put "map/set syntax" on the global hoon wishlist.)

Reply:
Yes, this is certainly on my radar.

Reply:
Also wait a second, `'cords'` don't support string interpolation! You need to `(crip "tape {etc}")` here.

Reply:
I don't like gas in map. It's less clear to me than my.